### ISSUE TITLE
User is sometimes unable to log in via WebUI when owning a SMS token (challenge response)

### ISSUE THREAD

**Issue by splashx** [privacyidea/privacyidea#995]
# What did you try to do?

1. Visited the WebUI
1. Entered username and password

# What outcome did you expect?

The system should show the challenge-response page (because of the policy set and tokens the user have) *with* a form for the user to enter the value of the code.

<img width="776" alt="screen shot 2018-04-03 at 16 29 52" src="https://user-images.githubusercontent.com/1154133/38265902-7cb11146-3777-11e8-9c37-984b0028026d.png">

# What outcome did you experience?

There were two outcomes (both missing the form):

1) The challenge response page did not have a form *and* didnt display the list of user tokens (all red cells except `F5` in the table at the end)

![screen shot 2018-04-03 at 18 44 10](https://user-images.githubusercontent.com/1154133/38265364-f2823848-3775-11e8-9a9c-b889d43f453b.png)

2) The challenge response page did not have a form *but* displayed the list of user token (`F5` in the table at the end) - **however** the list is incomplete

![screen shot 2018-04-03 at 18 46 26](https://user-images.githubusercontent.com/1154133/38265376-fc73fbb6-3775-11e8-8774-a1edfec2a7e8.png)

# Configuration

* privacyIDEA Version: v2.22

* Installation method: pip

## more details: 

* OS: Ubuntu 16.04

* Webserver: Apache

* Tokendatabase: MariaDB

# Log file

There is no log message (apart from the usual GET/POST webui login message) - this appears to be a user-side rendering issue.

# Extra

In our environment users can have:
1. `sms` only (transient state) *or*
1.  `sms` and `totp` token. 

We use the default value of `challenge_response` option in our authentication policy (which is empty+unchecked).

For debugging purposes we tried the combination of these variables and we noticed the behaviour differs: it depends on number of tokens the user has plus value of `challenge_response`. There is a summary below:

![screen shot 2018-04-03 at 18 51 38](https://user-images.githubusercontent.com/1154133/38265499-65cbf226-3776-11e8-959d-8a2a9fd12504.png)

**Comment by splashx** [comment]
>How does your authentication policy look like? Are you using otppin=userstore or otppin=token.

We use `otppin=userstore`. Below is a data which mimics the data used to configure the authentication policy in production:

```
{
  "action": [
    "passthru=userstore",
    "otppin=userstore"
  ],
  "scope": "authentication",
  "realm": [
    "company.eu"
  ],
  "resolver": [
    "ldap"
  ],
  "user": [],
  "active": true,
  "check_all_resolvers": false,
  "client": [],
  "time": "",
  "adminrealm": []
}
```

> Please note: .... 

Thanks for pointing out but we're aware of the behaviour: you can see D3 and D4 in the table are green and marked as "expected".

> If you say "transient state" should all users have an SMS and a TOTP token?

Correct - all users have SMS and TOTP tokens enrolled.

**Comment by fredreichbier** [comment]
I just merged #997 to ``branch-2.22``, so this should be fixed in the 2.22.1 release!
